Bottom: dbe2fa7fde61a7a78146698c81e5a935137615b5
Top:    9c6f7292840b0c93a7ba1293c24bce8e1d05f86a
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-02-17 22:49:59 +1300

Refresh of content-type-repr

---

diff --git a/NEWS b/NEWS
index 0c6b7f5..99895e9 100644
--- a/NEWS
+++ b/NEWS
@@ -11,6 +11,12 @@ NEXT
 
 A simple bug fix, and better error messages when you don't up-call.
 
+Changes
+-------
+
+* ``testtools.content_type.ContentType`` incorrectly used ',' rather than ';'
+  to separate parameters. (Robert Collins)
+
 Improvements
 ------------
 
diff --git a/testtools/content_type.py b/testtools/content_type.py
index c491408..bbf314b 100644
--- a/testtools/content_type.py
+++ b/testtools/content_type.py
@@ -29,7 +29,7 @@ class ContentType(object):
     def __repr__(self):
         if self.parameters:
             params = '; '
-            params += ', '.join(
+            params += '; '.join(
                 sorted('%s="%s"' % (k, v) for k, v in self.parameters.items()))
         else:
             params = ''
diff --git a/testtools/tests/test_content_type.py b/testtools/tests/test_content_type.py
index ecb8e3a..2d34f95 100644
--- a/testtools/tests/test_content_type.py
+++ b/testtools/tests/test_content_type.py
@@ -43,7 +43,7 @@ class TestContentType(TestCase):
         content_type = ContentType(
             'text', 'plain', {'foo': 'bar', 'baz': 'qux'})
         self.assertThat(
-            repr(content_type), Equals('text/plain; baz="qux", foo="bar"'))
+            repr(content_type), Equals('text/plain; baz="qux"; foo="bar"'))
 
 
 class TestBuiltinContentTypes(TestCase):
